### DISCUSSION TITLE
Not sure how to handle arrays of arrays...

### DISCUSSION THREAD

**Discussion by terebentina** [react-hook-form/react-hook-form#3337] [Q&A] [answered]
Here's the code sandbox: https://codesandbox.io/s/cranky-microservice-vt7qo?file=/src/App.js

I have a field array of "blocks" and each block can have a field array of "names".
The form starts with a default of 1 block with 1 name set as `defaultValues` in `useForm`.
```
{
  blocks: [{ nested: { names: [{ name: "old name" }] } }]
}
```
The problem I am facing is that if I:
1. remove the first block with `remove(0)` (the method coming from `useArrayField`)
2. then create a new block with `append({ nested: { names: [{ name: "new name" }] } })`

the old name is still displayed instead of the new one.
It's like the names field array doesn't unregister but I even tried manually unregistering on block unmount and it still behaves the same.
To reproduce click on `remove first block` then click on `add block`.

Appreciate any help!

**Comment by bluebill1049** [comment, marked as answer]
https://github.com/react-hook-form/react-hook-form/tree/master/examples

there is an example array of array, let me know if that helps.

**Reply by terebentina** [reply]
It does, thank you. 

I noticed that for the outer array you have an "append" button that does 
```
append({ name: "append" });
``` 
and "append nested" that does 
```
setValue("test", [
  ...getValues().test,
  {
    name: "append",
    nestedArray: [{ field1: "append", field2: "append" }]
  }
]);
```
My question is why isn't "append nested" doing 
```
append({
  name: "append",
  nestedArray: [{ field1: "append", field2: "append" }]
})
```
?

**Reply by bluebill1049** [reply]
because each `append` function belong to each `useFieldArray` hook.
